cdap-ui/app/cdap/components/Replicator/Detail/ThroughputGraph/parser.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { IThroughputData } from 'components/Replicator/Detail/ThroughputGraph/throughput';
+
+interface IMetricData {
+  time: number;
+  value: number;
+}
+
+interface IMetricSeries {
+  metricName: string;
+  data: IMetricData[];
+}
+
+interface IRawMetricData {
+  startTime: number;
+  endTime: number;
+  resolution: string;
+  series: IMetricSeries[];
+}
+
+const METRIC_NAME_MAP = {
+  'user.dml.insert': 'inserts',
+  'user.dml.update': 'updates',
+  'user.dml.delete': 'deletes',
+  'user.dml.error': 'errors',
+};
+
+const INITIAL_DATA = {
+  inserts: 0,
+  updates: 0,
+  deletes: 0,
+  errors: 0,
+};
+
+export function throughputParser(rawData: IRawMetricData): IThroughputData[] {
+  /**
+   * <time>: {
+   *    time: <value>,
+   *    inserts: <value>,
+   *    updates: <value>,
+   *    deletes: <value>,
+   *    errors: <value>
+   * }
+   */
+  const timeMap: Record<string, IThroughputData> = {};
+
+  rawData.series.forEach((metricSeries) => {
+    const metricType = METRIC_NAME_MAP[metricSeries.metricName];
+
+    metricSeries.data.forEach((metricData) => {
+      const time = metricData.time;
+      if (!timeMap[time]) {
+        timeMap[time] = {
+          time,
+          ...INITIAL_DATA,
+        };
+      }
+
+      timeMap[time][metricType] = metricData.value;
+    });
+  });
+
+  // Interpolate data with missing times
+  const resolution = parseInt(rawData.resolution, 10);
+  const startTime = rawData.startTime - (rawData.startTime % resolution);
+  let currentTime = startTime;
+  while (currentTime < rawData.endTime) {
+    if (!timeMap[currentTime]) {
+      timeMap[currentTime] = {
+        time: currentTime,
+        ...INITIAL_DATA,
```

Comment:
Would it be better to show this as `0` or as a break in the graph?

Reply:
Can you elaborate what you meant by showing as 0? Currently if there's no data, there is no bar on the graph for that time period

![image](https://user-images.githubusercontent.com/4398643/100802321-7e561f00-33e6-11eb-8f6a-d43277009eb1.png)

Reply:
When I originally read this part of the code, I assumed it was a line chart, not a bar chart. So leaving this aside, the only issue I can think of would be if there's a tooltip over `0` values and if that should be handled differently for missing data..